packages/agent/src/certificate.ts

```diff
@@ -99,59 +98,110 @@ function isBufferEqual(a: ArrayBuffer, b: ArrayBuffer): boolean {
   return true;
 }
 
+export interface CreateCertificateOptions {
+  /**
+   * The bytes encoding the certificate to be verified
+   */
+  certificate: ArrayBuffer;
+  /**
+   * The root key against which to verify the certificate
+   * (normally, the root key of the IC main network)
+   */
+  rootKey: ArrayBuffer;
+  /**
+   * The effective canister ID of the request that generated the certificate.
```

Comment:
The canister doesn't necessarily generate the certificate - the result can be just the call to `read_state` (for example, to read the status of a request). But you're right, it can also be generated by the canister. I think we can say effective canister ID/signing canister ID, depending on the context. I'll formulate something along those lines.